_deploy/r/gnoswap/common/tick_math.gno

```diff
@@ -43,34 +43,40 @@ var binaryLogConsts = [8]*u256.Uint{
 
 var (
 	shift1By32Left = u256.MustFromDecimal("4294967296") // (1 << 32)
+	maxTick        = int32(887272)
 )
 
 func TickMathGetSqrtRatioAtTick(tick int32) *u256.Uint { // uint160 sqrtPriceX96
 	absTick := abs(tick)
-	if absTick > 887272 { // MAX_TICK
+	if absTick > maxTick {
 		panic(addDetailToError(
 			errOutOfRange,
 			ufmt.Sprintf("tick_math.gno__TickMathGetSqrtRatioAtTick() || tick is out of range (larger than 887272), tick: %d", tick),
 		))
 	}
 
-	ratio := u256.MustFromDecimal("340282366920938463463374607431768211456") // consts.Q128
+	var initialBit int32 = 0x1
+	var ratio *u256.Uint
+	if (absTick & initialBit) != 0 {
+		ratio = tickRatioMap[initialBit]
+	} else {
+		ratio = u256.MustFromDecimal("340282366920938463463374607431768211456") // consts.Q128
+	}
 
 	for mask, value := range tickRatioMap {
-		if absTick&mask != 0 {
+		if (mask != initialBit) && absTick&mask != 0 {
```

Comment:
It would be better for readability to extract this conditional statement into a function and give it a name.